reactos/drivers/usb/usbhub/fdo.c

```diff
@@ -1404,8 +1436,15 @@ USBHUB_FdoQueryBusRelations(
 
     if (!DeviceRelations)
         return STATUS_INSUFFICIENT_RESOURCES;
```

Comment:
Actually, I was wrong here. Don't free the old device relations in this case. And don't touch IoStatus.Information. The PNP manager will free it on failure, correctly dereferencing all device objects in it.